### PR TITLE
Clear intergface for models

### PR DESCRIPTION
**Pull request by nicolas-chaulet**
As part of the effort towards a release as a standalone package, I wanted to make the code more modular. For example it is a bot tricky today to use the trackers since there not documentation of what the interface of the model should be. This PR tries to bring a bit more clarity into that.

### CODE REVIEW

File: torch_points3d/models/model_interface.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+
+class TrackerInterface(ABC):
+    @abstractmethod
+    def get_labels(self):
+        """ returns a trensor of size ``[N_points]`` where each value is the label of a point
+        """
+
+    @abstractmethod
+    def get_batch_idx(self):
```
Comment (humanpose1) [inline torch_points3d/models/model_interface.py:43]: I use the the name get_batch

File: torch_points3d/models/model_interface.py [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+        """
+
+    @abstractmethod
+    def get_output(self):
+        """ returns a trensor of size ``[N_points,...]`` where each value is the output
+        of the network for a point (output of the last layer in general)
+        """
+
+    @abstractmethod
+    def get_input(self):
```
Comment (humanpose1) [inline torch_points3d/models/model_interface.py:54]: I use the name get_xyz

File: torch_points3d/datasets/base_dataset.py [hunk 3]
```diff
@@ -7,6 +7,7 @@
 import torch_geometric
 from torch_geometric.transforms import Compose, FixedPoints
 
+from torch_points3d.models.model_interface import DatasetInterface
```
Comment (tchaton) [inline torch_points3d/datasets/base_dataset.py:10]: why modelInterface contains DatasetInterface ?

Comment (tchaton) [reply torch_points3d/datasets/base_dataset.py:10]: Ok. model_interface name seems a little ambiguos. Maybe we could find a better way to name it, as external_interfaces_to_model, or something.
or regrouping all of them within an torch_points3d.core.interfaces

File: torch_points3d/models/model_interface.py (hunk 2 above)
Comment (nicolas-chaulet) [reply torch_points3d/models/model_interface.py:54]: Get input could potentially contain the whole input data structure, it’s up to you

File: torch_points3d/models/model_interface.py (hunk 1 above)
Comment (nicolas-chaulet) [reply torch_points3d/models/model_interface.py:43]: Get batch is better! I’ll change that

File: torch_points3d/datasets/base_dataset.py (hunk 3 above)
Comment (nicolas-chaulet) [reply torch_points3d/datasets/base_dataset.py:10]: Yes, not sure what’s the best way to name those things.

Comment (nicolas-chaulet) [reply torch_points3d/datasets/base_dataset.py:10]: Another way to deal with it is to import model_interface and then use model_interface.DatasetInterface and so on. What do you think? In at case you need to reference model and dataset

Comment (nicolas-chaulet) [reply torch_points3d/datasets/base_dataset.py:10]: If they are all in the same file it will be even more confusing I think

Comment (tchaton) [reply torch_points3d/datasets/base_dataset.py:10]: Ok, let's keep it that way for now.

**Pull request merged**